### ISSUE TITLE
Inheritance and Inherited Members sometimes contain non-Core types

### ISSUE THREAD

**Issue by svick** [dotnet/core-docs#547]
The Inheritance and Inherited Members section for some types in API reference contain members that are not in .Net Core.

For example, this happens for [`FileStream`](https://dotnet.github.io/api/System.IO.FileStream.html): its Inheritance contains `MarshalByRefObject()` and its Inherited Members contain several incorrect members, including `Close()`.

I can't discern any pattern where this happens: other `Stream` types are affected too (e.g. [`NegotiateStream`](https://dotnet.github.io/api/System.Net.Security.NegotiateStream.html)), but not all (e.g. [`MemoryStream`](https://dotnet.github.io/api/System.IO.MemoryStream.html) is fine).

**Comment by mairaw** [comment]
Yes, it's showing the inheritance for the Framework somehow. Maybe something happened in the latest changes from docfx?

/cc @bradygaster, @chenkennt

**Comment by chenkennt** [comment]
@mairaw we recently added the inherited members to our default template as we think it's useful information to user. If it's not a desired behavior for .NET Core, we can remove that in .NET Core template.

**Comment by mairaw** [comment]
@chenkennt it's useful information and we do have bugs filed tracking the lack of that information. However, I don't think we ever discussed the design of this and it's showing the wrong information in some cases.

**Comment by mairaw** [comment]
@dend @bradygaster this is still an issue and we should discuss how we handle differences in hierarchy in the future when the content is merged. With .NET Standard 2.0, the difference might be smaller but we still need to know how to handle those.

**Comment by mairaw** [comment]
@dend how are we handling this?

**Comment by svick** [comment]
It looks like right now, it manages to be wrong on *both* .Net Core and .Net Framework:

![](https://cloud.githubusercontent.com/assets/287848/25506639/f227288e-2ba7-11e7-928d-9090489d6708.png)

The definition shows `Stream` inherits from `MarshalByRefObject`, which is wrong on .Net Core. The inheritance chain shows `Stream` inherits directly from `Object`, which is wrong on .Net Framework.

Also, the Inherited Members section for [`FileStream`](https://docs.microsoft.com/en-us/dotnet/api/system.io.filestream?view=netcore-1.1) incorrectly shows `Close()` on .Net Core.

And while the [.Net Core version of `Stream.Close()`](https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.close?view=netcore-1.1&viewFallbackFrom=netcore-1.1) correctly says it is not supported, clicking the `Close()` link from the .Net Core version of `FileStream` takes you to the .Net Framework version of `Stream.Close()`, which could be confusing (the URL says `view=netframework-4.7&viewFallbackFrom=netcore-1.1`).

**Comment by mairaw** [comment]
@joelmartinez @dend is this being tracked as part of future improvements to mdoc? Ability to show different inheritance, different signature, etc.?
